libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/junction/view/MapboxJunctionView.kt

```diff
@@ -42,33 +47,66 @@ class MapboxJunctionView : AppCompatImageView {
 
+    private val ioJobController: JobControl = ThreadController.getIOScopeAndRootJob()
+    private val mainJobController: JobControl = ThreadController.getMainScopeAndRootJob()
+    private var job: Job? = null
+
+    private val callback = object : MapboxNavigationConsumer<Bitmap?> {
+        override fun accept(value: Bitmap?) {
+            visibility = if (value != null) {
+                setImageBitmap(value)
+                VISIBLE
+            } else {
+                setImageBitmap(null)
+                GONE
+            }
+        }
+    }
+
     /**
-     * Invoke to render the signboard based on data or error conditions.
+     * Invoke the API to render the junction based on data or error conditions.
+     *
+     * @param result type [Expected] which holds either the junction or error
+     * @param junctionCallback an optional nullable parameter. If null, the SDK handles rendering
+     * the junction as [Bitmap] and controls the visibility. If not null, the SDK passes the [Bitmap]
+     * to the caller and allows them to decide when to render the [Bitmap] and control the visibility.
      */
-    fun render(result: Expected<JunctionValue, JunctionError>) {
+    @JvmOverloads
+    fun render(
+        result: Expected<JunctionValue, JunctionError>,
+        junctionCallback: MapboxNavigationConsumer<Bitmap?>? = null
+    ) {
         when (result) {
             is Expected.Success -> {
-                val signboard = renderJunction(result.value)
-                visibility = if (signboard != null) {
-                    setImageBitmap(signboard)
-                    VISIBLE
+                junctionCallback?.let {
+                    renderJunction(result.value, it)
+                } ?: renderJunction(result.value, callback)
+            }
+            is Expected.Failure -> {
+                if (junctionCallback != null) {
+                    junctionCallback.accept(null)
                 } else {
                     setImageBitmap(null)
-                    GONE
+                    visibility = GONE
                 }
             }
-            is Expected.Failure -> {
-                setImageBitmap(null)
-                visibility = GONE
-            }
         }
     }
 
-    private fun renderJunction(value: JunctionValue): Bitmap? {
+    private fun renderJunction(
+        value: JunctionValue,
+        callback: MapboxNavigationConsumer<Bitmap?>
+    ) {
         return when (value.bytes.isEmpty()) {
-            true -> null
+            true -> callback.accept(null)
             else -> {
-                BitmapFactory.decodeByteArray(value.bytes, 0, value.bytes.size)
+                job?.cancel()
+                job = ioJobController.scope.launch {
+                    val junction = BitmapFactory.decodeByteArray(value.bytes, 0, value.bytes.size)
+                    mainJobController.scope.launch {
+                        callback.accept(junction)
+                    }
+                }
```

Comment:
Let's align with the Signboard APIs and refactor this to something like:
```kotlin
    fun render(
        result: Expected<JunctionValue, JunctionError>
    ) {
        when (result) {
            is Expected.Success -> {
                setImageBitmap(result.value)
            }
            is Expected.Failure -> {
                setImageBitmap(null)
            }
        }
    }
```

Reply:
What do you think of handling this in a separate PR so it unblocks Signboard related work?
cc @LukasPaczos

Reply:
We can do that as well, but let's remove all junction-related changes from this PR then.

Reply:
Fixed the junctions as well @LukasPaczos